gtsam/geometry/Cal3Bundler.cpp

```diff
@@ -99,20 +99,26 @@ Point2 Cal3Bundler::calibrate(const Point2& pi, OptionalJacobian<2, 3> Dcal,
   double x = (pi.x() - u0_) / fx_, y = (pi.y() - v0_) / fx_;
   const Point2 invKPi(x, y);
 
-  // initialize by ignoring the distortion at all, might be problematic for
-  // pixels around boundary
-  Point2 pn(x, y);
+  Point2 pn;
+  double px = pi.x(), py = pi.y();
```

Comment:
Per our discussion in the other thread, this should be x, y right? I was basically suggesting the following: 

```
  double px = (pi.x() - u0_) / fx_, py = (pi.y() - v0_) / fx_;  
  const Point2 invKPi(px, py);  
  Point2 pn;
```
Then the do-while as you have below. 
It might be worth adding another unit test for a `Cal3Bundler K(5, 0, 0, 2, 2);` i.e non-zero `u0_`, `v0_`.

Reply:
Ah good catch. Sorry I missed that.